Singleton/version_1/Singleton.hpp

```diff
@@ -0,0 +1,31 @@
+# ifndef SINGLETON_HPP
+# define SINGLETON_HPP
+
+# include <iostream>
+# include <mutex>
+# include <thread>
+
+
+class Singleton
+{
+	public:
+		// void				setName(std::string const&);
+		std::string			getName() const;
+		static Singleton&	getInstacne(std::string const&);
+	
+	private:
+		Singleton(std::string const&);
+
+	public:
+		Singleton(Singleton const&) = delete;
+		Singleton(Singleton&&) = delete;
```

Comment:
Please check this article:

Reply:
Thank you for information, that was useful!
I'll follow your advice :)